### DISCUSSION TITLE
Adapter for MPP Solar Inverters

### DISCUSSION THREAD

**Discussion by softwarecrash** [arendst/Tasmota#11092] [Ideas]
Hello,
i, and sure many other people have this "Offgrid" solar inverters, brandet as PIP, MPP, SMV, Isolar .......
all invertes from esun are have the same communication
its a lot of devices see https://www.ebay.de/sch/easunpowerpro/m.html?_nkw=&_armrs=1&_ipg=&_from=
all devices with th RJ45 Jack marked as COM port near the USB port 
all this inverters can talk over a simple RS232 connection, its possibile to read data from this.

all what need is a rs232 to ttl adapter, then send a simple command "QPIGS", and the inverter answer with a long string like 
(AAA BB.B CCC DD.D.........
so the string explode is 
( start
AAA Grid Voltage
BB.B Grid Freq
CCC AC out Voltage
DD.D AC out Freq
... and more data
thats it.
note: for security i think only this command is enugh, other commands allow to setup the inverter so wrong settings can destroy someting

any interesting to develope a adapter for the solar inverter?
 i dont have enugh skills to do that for tasmota

here is the protocol
https://www.photovoltaikforum.com/core/attachment/7135-protocol-pdf/
the interesting query is the QPIGS and the answer

here is my working try
https://github.com/softwarecrash/InverterOfThings (i have made changes that not online but it show how it work - sorry for the chaos, i have forked that and only gett it to work)

i can test it and wrote a how to etc
or a working scelet of this that i cann fill it out with the right params
when it needet i can made a tiny script for a second esp or attiny to simulate the inverter or make the answer

PS: sorry i cant send you the inverter for development, i need it, realy! but a bounty for beer and pizza :)

**Comment by proasnet** [comment]
Use my AxpertinoGW light on github.
